### PR TITLE
test: swap flow cypress tests

### PR DESCRIPTION
**Pull request by just-toby**
## Description
updates the swap flow e2e tests - adds some test cases, and fixes and re-enables an existing test case.

also adds some e2e tests for the widget on TDP

https://uniswaplabs.atlassian.net/browse/WEB-3164?atlOrigin=eyJpIjoiYTlkZWI1ZTQ2NDE5NDYzNGI3ZGNlYWQ2MjA4ZjI0NDYiLCJwIjoiaiJ9

# test plan
### Automated testing

tests pass

### CODE REVIEW

File: cypress/e2e/token-details.test.ts [hunk 1]
```diff
@@ -1,4 +1,6 @@
-import { getTestSelector } from '../utils'
+import { getClassContainsSelector, getTestSelector } from '../utils'
+
+const UNI_GOERLI = '0x1f9840a85d5aF5bf1D1762F925BDADdC4201F984'
```
Comment (zzmp) [inline cypress/e2e/token-details.test.ts:3]: You should pull this from src/constants/addresses.ts

File: cypress/e2e/swap.test.ts [hunk 2]
```diff
@@ -1,13 +1,57 @@
+import { getTestSelector } from '../utils'
+
+const WETH_GOERLI = '0xB4FBF271143F4FBf7B91A5ded31805e42b2208d6'
```
Comment (zzmp) [inline cypress/e2e/swap.test.ts:3]: You should pull this from src/constants/addresses.ts

**Review by zzmp — APPROVED** [review]
lgtm, but the test constansts for token addresses should be pulled from src/constants/addresses.ts

File: cypress/e2e/token-details.test.ts (hunk 1 above)
Comment (just-toby) [reply cypress/e2e/token-details.test.ts]: this results in a build error in the test, so i'm leaving as is for now

File: cypress/e2e/swap.test.ts [hunk 3]
```diff
@@ -1,13 +1,57 @@
...
+    } else {
+      cy.get(`#swap-currency-${field} .token-symbol-container`).should('contain.text', tokenSymbol)
+    }
+  }
+
+  const selectOutput = (tokenSymbol: string) => {
+    // open token selector...
+    cy.contains('Select token').click()
+    // select token...
+    cy.contains(tokenSymbol).click({ force: true })
```
Comment (vm) [inline cypress/e2e/swap.test.ts:26]: why do you need force?

File: cypress/e2e/swap.test.ts [hunk 4]
```diff
@@ -1,13 +1,57 @@
...
+
+    cy.get('body')
+      .then(($body) => {
+        if ($body.find(getTestSelector('TokenSafetyWrapper')).length) {
+          return 'I understand'
+        }
+
+        return 'no-op' // Don't click on anything, a no-op
```
Comment (vm) [inline cypress/e2e/swap.test.ts:34]: why not return undefined here?

File: cypress/e2e/swap.test.ts [hunk 5]
```diff
@@ -1,13 +1,57 @@
...
+  const selectOutput = (tokenSymbol: string) => {
+    // open token selector...
+    cy.contains('Select token').click()
+    // select token...
+    cy.contains(tokenSymbol).click({ force: true })
+
+    cy.get('body')
+      .then(($body) => {
+        if ($body.find(getTestSelector('TokenSafetyWrapper')).length) {
+          return 'I understand'
```
Comment (vm) [inline cypress/e2e/swap.test.ts:31]: this is not a selector, its the text content

**Review by vm — APPROVED** [review]
some nits

File: cypress/e2e/swap.test.ts (hunk 5 above)
Comment (just-toby) [reply cypress/e2e/swap.test.ts:31]: i'll rename

**Review by vm — APPROVED** [review]

**Pull request merged**